tests/integration/ha/test_ha.py

```diff
@@ -443,21 +445,82 @@ async def test_full_cluster_crash(
     for unit_id in get_application_unit_ids(ops_test, app):
         await update_restart_delay(ops_test, app, unit_id, ORIGINAL_RESTART_DELAY)
 
-    # sleep for restart delay + 30 secs max for the election time + node start + cluster formation
-    time.sleep(ORIGINAL_RESTART_DELAY + 30)
+    # sleep for restart delay + 45 secs max for the election time + node start + cluster formation
+    # around 10 sec enough in a good machine - 45 secs for CI
+    time.sleep(ORIGINAL_RESTART_DELAY + 45)
 
     # verify all units are up and running
     for unit_id, unit_ip in get_application_unit_ids_ips(ops_test, app).items():
         assert await is_up(ops_test, unit_ip), f"Unit {unit_id} not restarted after cluster crash."
```

Comment:
if you wanted you could have a retry over the period `ORIGINAL_RESTART_DELAY + 45` - this might make your tests run faster. But up to you